src/core/stores/VDocument.ts

```diff
@@ -25,6 +25,29 @@ export class VDocument {
     // Public
     //--------------------------------------------------------------------------
 
+    /**
+     * Handle the general expected behavior (a block split) of the enter key.
+     */
+    enter(): void {
```

Comment:
Shall we call it `splitParagraph` and have the insert BR intent be triggered by the EventManager?

Reply:
Yes, it's how I envisioned it. The EventManager checks whether shift was pressed as well and triggers the correct intent (and thus the correct command) accordingly. Maybe we could even simplify this process later with the keymap if we create a keymap which binds `Enter` to `splitParagraph` and `SHIFT+ENTER` to `insertLineBreak`.

I'm still not 100% convinced of the name though. Is the intent of the user really to split something ? I'm not sure... I think the user just wants to insert a carriage return. It's an implement detail of the tree structure of HTML that "inserting a carriage return" into a node ends up actually splitting it. It even makes sense if you think about it as a typwriting machine (like all control characters and key terminology were inherited from):

You would have mechanically activated some formatting mode in your typewriter, let's call it "H1". You write some text, then you pull on the "carriage return and line feed" lever, you are at the beginning (carriage return) of the next line (line feed) but you didn't toggle the "H1" mode by mechanically pressing its key, so you are continuing to write an "H1". It's just a detail of HTML that it actually split the H1 at its end in this case and that you end up with two H1.

However, this analogy has an issue. The old debate of CR vs LF. Technically, CR gets you at the start of the current line, and LF gets you on the next line, so we would need "CRLF" to have the proper result. So `insertCarriageReturn` is *technically* wrong, but so are the OS platforms using only `CR` or `LF` instead of `CRLF`.

To somewhat stay close to the CRLF terminology while not sticking too much to it, maybe we could do `insertReturn` ? So we would have `insertReturn` and `insertLineBreak` but then at that point you'd be surprised that "inserting a return" ends up splitting the node, and I just went full circle at this point.

Our challenge for today is to find a name that does not include `split`, because I don't think that splitting is the intent of the end user, but still hints the developer that the node might will be split.

Right now, I have the following ideas: `insertParagraphBreak`, `breakParagraph`, `break`. I like the word "break" because it conveys the "split" idea while not forcing it. Since there is "line break", you understand that "paragraph break" is just another typing control mechanism. It just breaks at a different level, which is what we want IMO. However, `break` is too similar to "line break" so I think we have to keep the word "paragraph" to avoid any confusion. Of my early three propositions, only two remains:
- `insertParagraphBreak`:
    - PROS: obvious counterpart of `insertLineBreak. Exactly the intent of the user.
    - CONS: "insert" dilutes the link between "break" and "split". You again end up splitting something by inserting something into it. Isn't that weird ? At first glance I'd say yes. At second glance, I'm more "well, I inserted a *break*, I should not really be surprised that I actually *broke* it, should I ?"
- `breakParagraph`:
    - PROS: more direct, I'm obviously breaking this paragraph.
    - CONS: But again, is this what I wanted to do ? If I'm at the end of a paragraph, do I want to "break" it ? If I'm in a cell of a table, I press enter, does it "break" something ? I'm scared of actually *breaking*, in the sense of "destroying", some stuff by calling this command in my code.

Looking at the `inputType` attribute of `InputEvent`, they have `insertLineBreak` which is "insert a line break" and `insertParagraph` which is "insert a paragraph break". Notice how `insertParagraph` is missing the word "break" but they still had to use it in the description because it makes no sense without it ? I definitely chose my challenger for today and it is `insertParagraphBreak` ! 🥇

Reply:
I'd also be in favor of either `insertParagraphBreak` or something to do with the carriage return like `insertCarriageReturn` or `returnCarriage` or `crlf`.